modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/database/H2TreeIndex.java

```diff
@@ -136,6 +136,9 @@ public class H2TreeIndex extends H2TreeIndexBase {
     /** */
     private final GridMessageListener msgLsnr;
 
+    /** Warned about index column type mismatch. */
+    private boolean idxTypeMismatchWarn;
```

Comment:
What reason to have the flag as class field?

Reply:
To do warning only once. Already switched to log throttling.